apps/browser/src/autofill/services/autofill.service.ts

```diff
@@ -964,18 +964,20 @@ export default class AutofillService implements AutofillServiceInterface {
       fillScript.autosubmit = Array.from(formElementsSet);
     }
 
-    if (options.allowTotpAutofill) {
+    if (options.allowTotpAutofill && options.cipher.login.totp) {
       await Promise.all(
         totps.map(async (t, i) => {
           if (Object.prototype.hasOwnProperty.call(filledFields, t.opid)) {
             return;
           }
 
           filledFields[t.opid] = t;
-          const totpResponse = await firstValueFrom(
-            this.totpService.getCode$(options.cipher.login.totp),
-          );
-          let totpValue = totpResponse.code;
+
+          const totpResponse = options.cipher.login?.totp
+            ? await firstValueFrom(this.totpService.getCode$(options.cipher.login.totp))
+            : undefined;
+
+          let totpValue = totpResponse?.code;
```

Comment:
Since you handle guarding this at the block conditional, no need to handle this case here:

```suggestion
          const totpResponse = await firstValueFrom(
            this.totpService.getCode$(login.totp),
          );
          let totpValue = totpResponse.code;
```